tanner/session_analyzer.py

```diff
@@ -114,26 +112,14 @@ def choose_possible_owner(stats):
                       'Mozilla/5.0 (Windows Phone 8.1; ARM; Trident/7.0; Touch; rv:11.0; '
                       'IEMobile/11.0; NOKIA; Lumia 530) like Gecko (compatible; bingbot/2.0; '
                       '+http://www.bing.com/bingbot.htm)']
-        if stats['user_agent'] in bots_owner:
-            hostname, _, _ = socket.gethostbyaddr(stats['peer_ip'])
-            if 'search.msn.com' or 'googlebot.com' in hostname:
-                possible_owners['crawler'] += 1
-            else:
-                possible_owners['attacker'] += 1
-        if stats['requests_in_second'] >= 10:
-            possible_owners['tool'] += 1
-            possible_owners['crawler'] += 1
-        else:
-            possible_owners['user'] += 1
-            possible_owners['attacker'] += 1
-        if stats['hidden_links'] > 0:
-            possible_owners['crawler'] += 1
-            possible_owners['attacker'] += 1
-        if set(stats['attack_types']).intersection(attacks):
-            possible_owners['attacker'] += 1
+        possible_owners['crawler'], possible_owners['tool'] = await self.detect_crawler(stats, bots_owner)
```

Comment:
since you use bots_owner only in one function, it's better to move it into it instead of pass it as a parameter

Reply:
@afeena bots_owner is used in detect_attacker function also (please check line no.148). That's why I passed it as parameter for both the functions.

Reply:
My bad :)